using-mobile-extensions/adobe-journey-optimizer-decisioning/api-reference.md

```diff
@@ -0,0 +1,952 @@
+     *
+     * @return {@link Proposition} instance.
+     */
+    public Proposition getProposition() {...}
+
+    /**
+     * Dispatches an event for the Edge network extension to send an Experience Event to the Edge network with the display interaction data for the
+     * given {@code Proposition} offer.
+     */
+    public void displayed() {...}
+
+    /**
+     * Dispatches an event for the Edge network extension to send an Experience Event to the Edge network with the tap interaction data for the
+     * given {@code Proposition} offer.
+     */
+    public void tapped() {...}
+
+    /**
+     * Generates a map containing XDM formatted data for {@code Experience Event - Proposition Interactions} field group from this {@code Proposition} item.
+     * <p>
+     * The returned XDM data does contain the {@code eventType} for the Experience Event with value {@code decisioning.propositionDisplay}.
+     * <p>
+     * Note: The Edge sendEvent API can be used to dispatch this data in an Experience Event along with any additional XDM, free-form data, and override
+     * dataset identifier.
+     *
+     * @return {@code Map<String, Object>} containing the XDM data for the proposition interaction.
+     */
+    public Map<String, Object> generateDisplayInteractionXdm() {...}
+
+    /**
+     * Generates a map containing XDM formatted data for {@code Experience Event - Proposition Interactions} field group from this {@code Proposition} offer.
+     * <p>
+     * The returned XDM data contains the {@code eventType} for the Experience Event with value {@code decisioning.propositionInteract}.
+     * <p>
+     * Note: The Edge sendEvent API can be used to dispatch this data in an Experience Event along with any additional XDM, free-form data, and override
+     * dataset identifier.
+     *
+     * @return {@code Map<String, Object>} containing the XDM data for the proposition interaction.
+     */
+    public Map<String, Object> generateTapInteractionXdm() {...}
+
+}
+```
+{% endtab %}
+{% tab title="iOS (AEP 3.x)" %}
+
+#### Swift
+
+```swift
+/// `Offer` class
+@objc(AEPOffer)
+public class Offer: NSObject, Codable {
+    /// Unique Offer identifier
+    @objc public let id: String
+
+    /// Offer revision detail at the time of the request
+    @objc public let etag: String
+
+    /// Offer schema string
+    @objc public let schema: String
+
+    /// Offer type as represented in enum `OfferType`
+    @objc public let type: OfferType
+
+    /// Optional Offer language array
+    @objc public let language: [String]?
+
+    /// Offer content string
+    @objc public let content: String
+
+    /// Optional Offer characteristics dictionary
+    @objc public let characteristics: [String: String]?
+}
+```
+
+The `Offer` class extension provides methods for generating XDM data for Proposition Interactions field group which can be used for proposition tracking. It also contains direct methods for tracking proposition display and tap interactions.
```

Comment:
"... for generating XDM data for the proposition interactions field group, which can be used..."

Reply:
Proposition Interactions is the name of the field group, shouldn't we retain the case here?

Reply:
Oh it is? Then yes, it should retain the case then.